app.js

```diff
@@ -0,0 +1,185 @@
+adjustHealthBars(chosenMaxLife);
+
+const normalAttackValue = 10;
+const monsterAttackValue = 15;
+const strongAttackValue = 20;
+const defaultHealvalue = 22;
+
+function adjustHealthBars(maxLife) {
+  monsterHealthBar.max = maxLife;
+  monsterHealthBar.value = maxLife;
+  playerHealthBar.max = maxLife;
+  playerHealthBar.value = maxLife;
+}
+
+function dealMonsterDamage(damage) {
+  const dealtDamage = Math.random() * damage;
+  monsterHealthBar.value = +monsterHealthBar.value - dealtDamage;
+  return dealtDamage;
+}
+
+function dealPlayerDamage(damage) {
+  const dealtDamage = Math.random() * damage;
+  playerHealthBar.value = +playerHealthBar.value - dealtDamage;
+  return dealtDamage;
+}
+
+function increasePlayerHealth(healValue) {
+  playerHealthBar.value = +playerHealthBar.value + healValue;
+}
+
+function resetGame(value) {
+  playerHealthBar.value = value;
+  monsterHealthBar.value = value;
+}
+
+function removeBonusLife() {
+  bonusLifeEl.parentNode.removeChild(bonusLifeEl);
+}
+
+function setPlayerHealth(health) {
+  playerHealthBar.value = health;
+}
+
+function reset() {
+  playerHealth = chosenMaxLife;
+  monsterHealth = chosenMaxLife;
+  resetGame(chosenMaxLife);
+}
+
+function writeLog(ev, val, logMonsterHealth, logPlayerHealth) {
+  let logEntry = {
+    event: ev,
+    value: val,
+    finalMonsterHealth: logMonsterHealth,
+    finalPlayerHealth: logPlayerHealth,
+  };
+
+  if (ev === logEventPlayerAttack || ev === logEventStrongAttack) {
+    logEntry.target = "MONSTER";
+  } else if (ev === logEventMonsterAttack) {
+    logEntry.target = "PLAYER";
+  } else if (ev === logEventPlayerHeal) {
+    logEntry = {
+      event: ev,
+      value: val,
+      target: "PLAYER",
+      finalMonsterHealth: logMonsterHealth,
+      finalPlayerHealth: logPlayerHealth,
+    };
+  } else if (ev === logEventGameOver) {
+    logEntry = {
+      event: ev,
+      value: val,
+      finalMonsterHealth: logMonsterHealth,
+      finalPlayerHealth: logPlayerHealth,
+    };
+  }
+  battleLog.push(logEntry);
+}
+
+function endRound() {
+  const initialPlayerHealth = playerHealth;
+  const playerDamage = dealPlayerDamage(monsterAttackValue);
+  playerHealth -= playerDamage;
+  writeLog(logEventMonsterAttack, playerDamage, monsterHealth, playerHealth);
+
+  if (playerHealth <= 0 && hasBonusLife) {
+    hasBonusLife = false;
+    removeBonusLife();
+    playerHealth = initialPlayerHealth;
+    setPlayerHealth(initialPlayerHealth);
+    alert("You would be dead but the bonus life saved you!");
+  }
+
+  if (monsterHealth <= 0 && playerHealth > 0) {
+    alert("You won!");
+    reset();
+    writeLog(logEventGameOver, "Player Won!", monsterHealth, playerHealth);
+  } else if (playerHealth <= 0 && monsterHealth > 0) {
+    alert("You lost!");
+    reset();
+    writeLog(logEventGameOver, "Monster Won!", monsterHealth, playerHealth);
+  } else if (playerHealth <= 0 && monsterHealth <= 0) {
+    alert("You have a draw!");
+    reset();
```

Comment:
Is this function being called? @Abbas4tech

Reply:
Yes!!! @Tauqeer-Ahmed-99